src/OpenTelemetry.Instrumentation.Wcf/Implementation/AspNetParentSpanCorrector.cs

```diff
@@ -0,0 +1,151 @@
+// <copyright file="AspNetParentSpanCorrector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.Instrumentation.Wcf.Implementation;
+
+/// <summary>
+/// When WCF is hosted in ASP.NET and the ASP.NET telemetry instrumentation is installed, the execution context does not flow from ASP.NET into WCF
+/// so the span which is created by ASP.NET is not visible to WCF. When OpenTelemetry.Instrumentation.Wcf then creates its own span it uses the parent
+/// which is passed from the caller. This results in the ASP.NET span and the WCF span being siblings off the same parent when, really, the WCF span
+/// should actually be a child of the ASP.NET (transport) span. This class corrects that behavior so the generated spans are correctly parented.
+///
+/// The way it does that is it hooks into the ASP.NET telemetry and rewrites the incoming request headers to reflect the ASP.NET parent (so, as far
+/// as WCF is concerned, it appears like the ASP.NET parent span was sent from the caller). It does this all via reflection to avoid having explicit
+/// dependencies on System.Web and OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule. This way this behavior is only enabled if the consumer
+/// also has the ASP.NET instrumentation installed.
+/// </summary>
+internal static class AspNetParentSpanCorrector
+{
+    private const string TelemetryHttpModuleTypeName = "OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule, OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule, PublicKeyToken=7bd6737fe5b67e3c";
+    private const string TelemetryHttpModuleOptionsTypeName = "OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModuleOptions, OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule, PublicKeyToken=7bd6737fe5b67e3c";
```

Comment:
As I remember usually we handle two cases (at least for internal tests visiblity) - with and withoud public key. Depending on library has this key or no.

Reply:
I couldn't actually find anywhere referencing other assemblies using their strong name, so I've updated this one to be consistent with the others and just not include the PublicKeyToken at all.